package/lftp: Build fails

On an attempt to build the `lftp-4.9.2`, the process fails with:
```
lftp_ssl.cc: In static member function ‘static int lftp_ssl_openssl::verify_crl(X509_STORE_CTX*)’:
lftp_ssl.cc:1164:10: error: ‘X509_STORE_CTX_get_by_subject’ was not declared in this scope; did you mean ‘X509_STORE_get_by_subject’?
 1164 |     rc = X509_STORE_CTX_get_by_subject(store_ctx, X509_LU_CRL, subject, obj);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          X509_STORE_get_by_subject
make[4]: *** [Makefile:2370: liblftp_network_la-lftp_ssl.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
ftpclass.cc: In member function ‘void Ftp::CheckResp(int)’:
ftpclass.cc:4377:16: warning: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘union sockaddr_u’; use assignment or value-initialization instead [-Wclass-memaccess]
 4377 |          memset(&conn->data_sa,0,sizeof(conn->data_sa));
      |          ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from Resolver.h:27,
                 from NetAccess.h:24,
                 from ftpclass.h:26,
                 from ftpclass.cc:31:
network.h:67:7: note: ‘union sockaddr_u’ declared here
   67 | union sockaddr_u
      |       ^~~~~~~~~~
```

The only place on the net that I found with the same error was one of [buildroot's autobuilds itself](http://autobuild.buildroot.org/results/572/572e32f80237e8333a62fc76612c255489d661c5/build-end.log). The buildroot HEAD is [package/uftrace: bump to version 0.10](https://git.buildroot.net/buildroot/commit/?id=d4877e6f88) ([github](https://github.com/buildroot/buildroot/commit/d4877e6f88)) and here comes the configurations:

```
BR2_arcle=y
BR2_archs38=y
BR2_ENABLE_DEBUG=y
BR2_DEBUG_3=y
BR2_TOOLCHAIN_EXTERNAL=y
BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
BR2_TOOLCHAIN_EXTERNAL_PATH="/install/arcgnu/gl_bingdb"
BR2_TOOLCHAIN_EXTERNAL_HEADERS_5_14=y
BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
BR2_TOOLCHAIN_EXTERNAL_HAS_SSP_STRONG=y
# BR2_TOOLCHAIN_EXTERNAL_INET_RPC is not set
BR2_TOOLCHAIN_EXTERNAL_CXX=y
BR2_TARGET_GENERIC_HOSTNAME="arcv2"
BR2_ROOTFS_DEVICE_TABLE="/tools/buildroot/device_table.txt"
BR2_SYSTEM_DHCP="eth0"
BR2_ROOTFS_OVERLAY="/tools/buildroot/overlay/qemu"
BR2_PACKAGE_STRACE=y
BR2_PACKAGE_LIBRESSL=y
BR2_PACKAGE_NCURSES_WCHAR=y
BR2_PACKAGE_NCURSES_TARGET_PROGS=y
BR2_PACKAGE_NCURSES_ADDITIONAL_TERMINFO="t/tmux"
BR2_PACKAGE_LFTP=y
BR2_PACKAGE_OPENSSH=y
BR2_TARGET_ROOTFS_CPIO=y
BR2_PACKAGE_HOST_KMOD=y
```

--- a/package/lftp/0001-Fix-build-with-LibreSSL-following-commit-537f37898.patch
+++ b/package/lftp/0001-Fix-build-with-LibreSSL-following-commit-537f37898.patch
@@ -1,0 +1,43 @@
+From 0276d5c239c41b3a63f738a5dc47fd56ac389cf0 Mon Sep 17 00:00:00 2001
+From: Ganael Laplanche <ganael.laplanche@martymac.org>
+Date: Thu, 20 Aug 2020 12:29:05 +0200
+Subject: [PATCH] Fix build with LibreSSL (following commit 537f37898)
+
+---
+ src/lftp_ssl.cc | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/src/lftp_ssl.cc b/src/lftp_ssl.cc
+index 14a3b9d4..968d3fb2 100644
+--- a/src/lftp_ssl.cc
++++ b/src/lftp_ssl.cc
+@@ -34,7 +34,7 @@
+ #include "misc.h"
+ #include "network.h"
+ #include "buffer.h"
+-#if OPENSSL_VERSION_NUMBER < 0x10100000L
++#if OPENSSL_VERSION_NUMBER < 0x10100000L || LIBRESSL_VERSION_NUMBER
+ #define X509_STORE_CTX_get_by_subject X509_STORE_get_by_subject
+ #endif
+ extern "C" {
+@@ -840,7 +840,7 @@ lftp_ssl_openssl_instance::lftp_ssl_openssl_instance()
+    ssl_ctx=SSL_CTX_new();
+    X509_set_default_verify_paths(ssl_ctx->cert);
+ #else
+-#if OPENSSL_VERSION_NUMBER < 0x10100000L
++#if OPENSSL_VERSION_NUMBER < 0x10100000L || LIBRESSL_VERSION_NUMBER
+    SSLeay_add_ssl_algorithms();
+ #endif
+    ssl_ctx=SSL_CTX_new(SSLv23_client_method());
+@@ -1080,7 +1080,7 @@ void lftp_ssl_openssl::copy_sid(const lftp_ssl_openssl *o)
+ 
+ const char *lftp_ssl_openssl::strerror()
+ {
+-#if OPENSSL_VERSION_NUMBER < 0x10100000L
++#if OPENSSL_VERSION_NUMBER < 0x10100000L || LIBRESSL_VERSION_NUMBER
+    SSL_load_error_strings();
+ #endif
+    int error=ERR_get_error();
+-- 
+2.25.1
+

